interfaces/builtin/common.go

```diff
@@ -162,8 +163,14 @@ func (iface *commonInterface) SecCompConnectedPlug(spec *seccomp.Specification,
 }
 
 func (iface *commonInterface) UDevConnectedPlug(spec *udev.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
-	for _, rule := range iface.connectedPlugUDev {
-		spec.TagDevice(rule)
+	// don't tag devices if the interface controls it's own device cgroup
+	if iface.controlsDeviceCgroup {
+		spec.SetControlsDeviceCgroup()
+	} else {
+		for _, rule := range iface.connectedPlugUDev {
+			spec.TagDevice(rule)
+		}
 	}
+
```

Comment:
It would be good to add tests to common_test.go like we did for suppressHomeIx.

Reply:
Done